src/App.tsx

```diff
@@ -1,14 +1,78 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [hasError, sethasError] = useState(false);
+  const [todosFilter, setTodosFilter] = useState<'completed' | 'active' | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const onClosed = () => {
+    setSelectedTodo(null);
+  };
+
+  const showAllTodos = () => {
+    setTodosFilter(null);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then((res) => {
+        setLoading(false);
+        setTodos(res);
+      })
+      .catch(() => {
+        sethasError(true);
+      });
+  }, []);
+
+  const showActive = () => {
+    setTodosFilter('active');
+  };
+
+  const showCompleted = () => {
+    setTodosFilter('completed');
+  };
```

Comment:
```suggestion
```

looks like these functions redundant.
consider to pass setTodosFilter function to TodoFilter component and just